src/seq.rs

```diff
@@ -257,8 +257,8 @@ mod test {
         let mut r = ::test::rng(402);
 
         // sample 0 items
-        assert_eq!(&sample_slice(&mut r, empty, 0)[..], []);
-        assert_eq!(&sample_slice(&mut r, &[42, 2, 42], 0)[..], []);
+        assert_eq!(&sample_slice(&mut r, empty, 0)[..], [0u8; 0]);
+        assert_eq!(&sample_slice(&mut r, &[42, 2, 42], 0)[..], [0u8; 0]);
```

Comment:
Is the extra verbosity useful?

Reply:
I don't know, this may seem a rust type inference bug.

![screenshot_20180227_224051](https://user-images.githubusercontent.com/6286761/36734921-df6319de-1c0f-11e8-920f-07ce6f78d09a.png)

Reply:
Try removing the first `&`; the function returns a `Vec` so it's not necessary.

Reply:
It does not work.

![screenshot_20180227_231025](https://user-images.githubusercontent.com/6286761/36736422-88313cdc-1c13-11e8-9c02-026983c96741.png)